app/components/agenda/agenda-header/agenda-actions.js

```diff
@@ -193,6 +198,48 @@ export default class AgendaAgendaHeaderAgendaActions extends Component {
     await this.decisionPublicationActivity.save();
   }
 
+  loadReports = task(async () => {
+    const INTERN_OVERHEID = await this.store.findRecordByUri(
+      'concept',
+      CONSTANTS.ACCESS_LEVELS.INTERN_OVERHEID,
+    );
+    const PUBLIEK = await this.store.findRecordByUri(
+      'concept',
+      CONSTANTS.ACCESS_LEVELS.PUBLIEK,
+    );
+    this.allReports = await this.store.queryAll('report', {
+      'filter[:has-no:next-piece]': true,
+      'filter[decision-activity][treatment][agendaitems][agenda][created-for][:id:]':
+        this.args.meeting.id,
+      sort: 'name',
+    });
+    this.signedReports = await this.store.queryAll('report', {
+      'filter[:has-no:next-piece]': true,
+      'filter[:has:signed-piece-copy]': true,
+      'filter[access-level][:id:]': [INTERN_OVERHEID.id, PUBLIEK.id].join(','),
+      'filter[decision-activity][treatment][agendaitems][agenda][created-for][:id:]':
+        this.args.meeting.id,
+      sort: 'name',
+    });
+  });
+
```

Comment:
Two remarks here (this was not clear in the story, so understandable that it wasn't done).

1. both of these need to be filtered on INTERN_OVERHEID
2. This check would be better in de backend.

In fact, as was perhaps not clear in the story, we don't even need the unsigned versions, just the flattened.

Reply:
> both of these need to be filtered on INTERN_OVERHEID

Double check with Johan. The initial implementation only had 1 button to generate a bundle of reports (up to, so including `PUBLIEK`) `INTERN_OVERHEID`, but Johan told me that the users really want 2 things:

- A bundle that only contains up to `INTERN_OVERHEID` signed (flattened) BFs, which is the thing they currently upload in Kaleidos
- A bundle that contains all unsigned BFs (including `VERTROUWELIJK` and other access levels) that the secretary uses during the meeting as a reference. And they have to be unsigned versions because the signing happens after the meeting IIRC.

> This check would be better in de backend.

Yes, but I added it here to keep the API of the service somewhat equivalent (send a list of reports to be bundled/generated together). But it's fine if the report collection is moved to the backend

Reply:
> * eting as a reference. And they have to be unsigned versions because the signing happens after the meeting IIRC.

@tdn This does not match our refactor...

Reply:
> > both of these need to be filtered on INTERN_OVERHEID
> 
> Double check with Johan. The initial implementation only had 1 button to generate a bundle of reports (up to, so including `PUBLIEK`) `INTERN_OVERHEID`, but Johan told me that the users really want 2 things:
> 
>     * A bundle that only contains up to `INTERN_OVERHEID` signed (flattened) BFs, which is the thing they currently upload in Kaleidos
> 
>     * A bundle that contains all unsigned BFs (including `VERTROUWELIJK` and other access levels) that the secretary uses during the meeting as a reference. And they have to be unsigned versions because the signing happens after the meeting IIRC.
> 
> 
> > This check would be better in de backend.
> 
> Yes, but I added it here to keep the API of the service somewhat equivalent (send a list of reports to be bundled/generated together). But it's fine if the report collection is moved to the backend

@tdn our refactor does not support the second bundle

Reply:
We decided to postpone the 2nd functionality due to it being too unclear and risky at the moment.
For example, other users than secretarie would also see this file, but cannot access it. 

With the current priorities in mind, this shouldn't have made it to implementation. Sorry for wasting your time with this!